solr/core/src/java/org/apache/solr/update/processor/AtomicUpdateDocumentMerger.java

```diff
@@ -148,6 +155,49 @@ public SolrInputDocument merge(final SolrInputDocument fromDoc, SolrInputDocumen
     return toDoc;
   }
 
+  private void doAddMerge(SolrInputDocument toDoc, SolrInputField sif, Object fieldVal) {
+    final String name = sif.getName();
+    SolrInputField existingField = toDoc.get(sif.getName());
+
+    if (existingField == null) {
+      doAdd(toDoc, sif, fieldVal);
+      return;
+    }
+
+    Map<String, SolrInputDocument> originalItemsById = existingField.getValues().stream()
```

Comment:
If the existing field doesn't contain values of type SolrInputDocument (or those don't have an id) better to throw an exception rather than ignore them.